src/mtcp/mtcp_restart.c

```diff
@@ -243,8 +244,12 @@ MTCP_PRINTF("Attach for debugging.");
     }
     // This assumes that the MTCP header signature is unique.
     do {
-      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
-    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+      rc = mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
+    } while (rc > 0 && mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+    if (rc == 0) { /* if end of file */
+      MTCP_PRINTF("***ERROR: ckpt image doesn't match MTCP_SIGNATURE\n");
```

Comment:
I wonder if we should rename `MTCP_SIGNATURE` with `DMTCP_SIGNATURE` in the error message so as to not confuse the user with MTCP/DMTCP.

We should keep this in mind for other such error strings as well.